htdocs/web_portal/controllers/downtime/add_downtime.php

```diff
@@ -71,53 +72,52 @@ function submit(\User $user = null) {
     if($confirmed == true){
         //Downtime is confirmed, submit it
         //$downtimeInfo = unserialize($_REQUEST['newValues']);  // didn't cater for UTF-8 chars
+        /**
+         * If confirmed by an user, submit the details of affected services
+         * and endpoints along with other details for each individual site.
+         */
         $downtimeInfo = json_decode($_POST['newValues'], TRUE);
         $serv = \Factory::getDowntimeService();
+        $downtimeInfo = unsetVariables($downtimeInfo, 'add');
+        $params = [];
+
+        foreach ($downtimeInfo['SITE_LEVEL_DETAILS'] as $siteID) {
+            $downtimeInfo['Impacted_Services'] = $siteID['services'];
+            $downtimeInfo['Impacted_Endpoints'] = $siteID['endpoints'];
+
+            $params['submittedDowntimes'][$siteID['siteName']] =
+                $serv->addDowntime($downtimeInfo, $user);
+        }
 
-        $params['dt'] = $serv->addDowntime($downtimeInfo, $user);
         show_view("downtime/added_downtime.php", $params);
     }else{
         //Show user confirmation screen with their input
         $downtimeInfo = getDtDataFromWeb();
 
-        //Need to sort the impacted_ids into impacted services and impacted endpoints
-        $impactedids = $downtimeInfo['IMPACTED_IDS'];
+        list(
+            $siteLevelDetails,
+            $serviceWithEndpoints
+        ) = endpointToServiceMapping($downtimeInfo['IMPACTED_IDS']);
 
-        $services=array();
-        $endpoints=array();
+        // Delete the unsorted IDs from the downtime info
+        unset($downtimeInfo['IMPACTED_IDS']);
 
-        //For each impacted id sort between endpoints and services using the prepended letter
-        foreach($impactedids as $id){
-            if (strpos($id, 's') !== FALSE){
-                //This is a service id
-                $services[] = str_replace('s', '', $id); //trim off the identifying char before storing in array
-            }else{
-                //This is an endpoint id
-                $endpoints[] = str_replace('e', '', $id); //trim off the identifying char before storing in array
-            }
+        if (!count($siteLevelDetails) > 1) {
+            $downtimeInfo['SINGLE_TIMEZONE'] = true;
```

Comment:
This is checking if more than one site has been selected, rather than if the sites invovled are in single timezone or not?

i.e. if a two site downtime was being added with both sites in UTC, `$downtimeInfo['SINGLE_TIMEZONE']` would not be set to true?

Reply:
Done in 949e465aaef13f1fe5bfe8f94dbd9534a25b3c32